indent/python.vim

```diff
@@ -205,13 +216,34 @@ function! s:indent_like_previous_line(lnum)
     let ignore_last_char = eval(s:skip_special_chars)
 
     " Search for final colon that is not inside a string or comment.
-    while search(':\s*\%(#.*\)\?$', 'bcW', lnum)
+    " It looks for the previous ":", and then checks if everything after it
+    " can be ignored.
+    let check_end = len(text)
+    let colon_pos = -1
+    while search(':', 'bcW', lnum)
       if eval(s:skip_special_chars)
         normal! h
-      else
-        return base + s:sw()
+        continue
+      endif
+      let colon_pos = getpos(".")[2]
+      " Check that everything behind is to be ignored, including jedi-vim's
+      " call signatures).
```

Comment:
here’s an extra paren.  maybe you could add a sentence explaining what those are?